gencred/cmd/main.go

```diff
@@ -0,0 +1,219 @@
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	flag "github.com/spf13/pflag"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/tools/clientcmd/api/latest"
+	gencred "k8s.io/test-infra/gencred/pkg"
+	"sigs.k8s.io/yaml"
+)
+
+const (
+	// defaultContextName is the default context name.
+	defaultContextName = "build"
+	// defaultConfigFileName is the default kubeconfig filename.
+	defaultConfigFileName = "/dev/stdout"
+)
+
+// options are the available command-line flags.
+type options struct {
+	context        string
+	name           string
+	output         string
+	certificate    bool
+	serviceaccount bool
+	overwrite      bool
+}
+
+// exitError is a custom error type which stores a message and status code.
+type exitError struct {
+	code    int
+	message string
+}
+
+func (err exitError) Error() string {
+	return err.message
+}
+
+// printErrAndExit prints an error message to stderr and exits with a status code.
+func printErrAndExit(err error) {
+	_, _ = fmt.Fprintln(os.Stderr, err.Error())
+
+	exitErr, ok := err.(*exitError)
+	if ok {
+		os.Exit(exitErr.code)
+	} else {
+		os.Exit(1)
+	}
+}
+
+// parseFlags parses the command-line flags.
+func (o *options) parseFlags() {
+	// --context
+	flag.StringVar(&o.context, "context", "", "The name of the kubeconfig context to use.")
+
+	// -n, --name
+	flag.StringVarP(&o.name, "name", "n", defaultContextName, "Context name for the kubeconfig entry.")
+
+	// -o, --output
+	flag.StringVarP(&o.output, "output", "o", defaultConfigFileName, "Output path for generated kubeconfig file.")
+
+	// -c, --certificate
+	flag.BoolVarP(&o.certificate, "certificate", "c", false, "Authorize with a client certificate and key.")
+
+	// -s, --serviceaccount
+	flag.BoolVarP(&o.serviceaccount, "serviceaccount", "s", false, "Authorize with a service account.")
+
+	// --overwrite
+	flag.BoolVar(&o.overwrite, "overwrite", false, "Overwrite (rather than merge) output file if exists.")
+
+	flag.Parse()
+}
+
+// validateFlags validates the command-line flags.
+func (o *options) validateFlags() error {
+	var err error
+
+	if len(o.name) == 0 {
+		return &exitError{message: "-n, --name option is required.", code: 1}
```

Comment:
Can we just use `--name`? Excessive verbosity decreases readability. Anyone using `-n` should already know it is `--name` and we should simply document the more readable `--name`

Reply:
I prefer to be *explicit* where possible. 4 extra character is **not** *overly* verbose IMO.